### PR TITLE
dash/highlight-sync-affected-series

### PR DESCRIPTION
**Pull request by vazonik**
Added `affectedSeriesId` option to the Highcharts Component's highlight sync.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206301298491242

### CODE REVIEW

File: docs/dashboards/synchronize-components.md [hunk 1]
```diff
@@ -130,6 +130,7 @@ Highlight sync can have additional options:
 sync: {
     highlight: {
         enabled: true,
+        affectedSeriesId: 'series-1',
```
Comment (karolkolodziej) [inline docs/dashboards/synchronize-components.md:133]: Please briefly describe this option.

File: ts/Dashboards/Components/HighchartsComponent/HighchartsComponentOptions.ts [hunk 2]
```diff
@@ -270,6 +270,18 @@ export interface SyncOptions extends Sync.RawOptionsRecord {
  * ```
  */
 export interface HighchartsHighlightSyncOptions extends Sync.OptionsEntry {
+    /**
+     * ID of the series that should be affected by the highlight. If not
+     * defined, the appropriate series will be found according to the column
+     * assignment.
```
Comment (karolkolodziej) [inline ts/Dashboards/Components/HighchartsComponent/HighchartsComponentOptions.ts:276]: Maybe worth mentioning that it makes only sense with `tooltip.shared: false`?

File: ts/Dashboards/Components/HighchartsComponent/HighchartsSyncs/HighchartsHighlightSync.ts [hunk 3]
```diff
@@ -224,13 +227,18 @@ const syncPair: Sync.SyncPair = {
...
 
-            if (chart && chart.series?.length) {
-                const cursor = e.cursor;
-                if (cursor.type === 'position') {
-                    let series;
+            if (chart && chart.series?.length && cursor.type === 'position') {
+                let series: Series | undefined;
+
+                if (highlightOptions.affectedSeriesId) {
+                    series = chart.get(
```
Comment (karolkolodziej) [inline ts/Dashboards/Components/HighchartsComponent/HighchartsSyncs/HighchartsHighlightSync.ts:234]: If my `affectedSeriesId`  is wrong and we fail to get the correct series, the sync won't work.
I see a few options to solve this: 
a) add a warning if no series is found
b) pick the first available series (as it works by default) 
c) both above?

File: docs/dashboards/synchronize-components.md (hunk 1 above)
Comment (sebastianbochan) [reply docs/dashboards/synchronize-components.md:133]: Just think, how about `relatedSeries` or `linkedSeriesId` names.

IMO `affected` series does not sound intuitive, however, it's only my impression.

File: samples/dashboards/sync/highcharts-highlight-affected-series/demo.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        }
+    }]
+});
+
+
+/**
+ * Custom code
+ **/
+
+const optionsCbx = document.querySelectorAll('.option');
```
Comment (sebastianbochan) [inline samples/dashboards/sync/highcharts-highlight-affected-series/demo.js:80]: Can we replace `Cbx` with something more meaningful?

Comment (Vazonik) [reply samples/dashboards/sync/highcharts-highlight-affected-series/demo.js:80]: Oh, right, I copied it from my old demo with checkboxes and didn't update the name even though there are radio buttons here... 🙈

**Review by karolkolodziej — APPROVED** [review]

**Review by stitot — APPROVED** [review]
Looks good to me

**Review by jomar-honsi — APPROVED** [review]

File: docs/dashboards/synchronize-components.md [hunk 5]
```diff
@@ -77,26 +78,18 @@ Dashboards.board('container', {
...
-      columnAssignment: {
-          Food: 'x',
-          'Vitamin A': 'value'
-      }
+      type: 'Highcharts'
   }, {
       renderTo: 'dashboard-col-1',
       connector: {
-          id: 'Vitamin'
+          id: 'Vitamin',
```
Comment (sebastianbochan) [inline docs/dashboards/synchronize-components.md:85]: unnecessary comma

**Review by sebastianbochan — APPROVED** [review]

**Pull request merged**